app/models/feed.rb

```diff
@@ -78,13 +77,18 @@ def sanitize_text
   end
 
   def set_public_feeds_url
-    if public? && url.blank? && ENV["PUBLIC_FEEDS_URL_PREFIX"].present?
-      self.url = "#{ENV["PUBLIC_FEEDS_URL_PREFIX"]}/#{podcast.path}/#{published_path}"
-    end
-  end
+    if private?
+      self.url = nil
+    elsif ENV["PUBLIC_FEEDS_URL_PREFIX"].present? && podcast.present?
+      public_feeds_url = "#{ENV["PUBLIC_FEEDS_URL_PREFIX"]}/#{path}"
 
-  def remove_url
-    self.url = nil
+      # if already publicfeeds... keep file_name/slug changes in sync
+      self.url = public_feeds_url if url.present? && url.include?(ENV["PUBLIC_FEEDS_URL_PREFIX"])
+
+      # otherwise, just default back to publicfeeds when blank
+      # TODO: after https://github.com/PRX/feeder.prx.org/issues/896 remove the date condition
+      self.url = public_feeds_url if url.blank? && (new_record? || created_at >= "2023-10-01")
```

Comment:
so the cases where `self.url` would not be changed are:

- new record, but the `url` has been filled in
- not new record, but it's not using the `PUBLIC_FEEDS_URL_PREFIX`

because we would want those `url`s to be whatever the user has set them to in those cases?
so number 2 is a case where `PUBLIC_FEEDS_URL_PREFIX` is present (L82) but is not used in the `url` (L86)?

Reply:
Exactly - we want to push users to _either_ use their own domain (like `https://feed.songexploder.net/SongExploder`) ... or use the `publicfeeds.net` url we generate for them.  We _don't_ want them to use `f.prxu.org` as the public feed url.

Reply:
but this logic doesn't change the url if they still chose to use `f.prxu.org`? like the option still needs to exist even though it's not encouraged?